driver/userspace.c

```diff
@@ -413,16 +504,16 @@ WnbdDeleteConnection(PGLOBAL_INFORMATION GInfo,
         TargetIndex = ScsiInfo->TargetIndex;
         BusIndex = ScsiInfo->BusIndex;
         ScsiInfo->SoftTerminateDevice = TRUE;
-        KeSetEvent(&ScsiInfo->DeviceEvent, (KPRIORITY)0, FALSE);
+        KeReleaseSemaphore(&ScsiInfo->DeviceEvent, 0, 1, FALSE);
         LARGE_INTEGER Timeout;
-        Timeout.QuadPart = (-1 * 1000 * 10000);
-        KeWaitForSingleObject(ScsiInfo->DeviceThread, Executive, KernelMode, FALSE, &Timeout);
-        if (-1 != ScsiInfo->Socket) {
-            WNBD_LOG_INFO("Closing socket FD: %d", ScsiInfo->Socket);
-            Close(ScsiInfo->Socket);
-            ScsiInfo->Socket = -1;
-        }
-        ObDereferenceObject(ScsiInfo->DeviceThread);
+        Timeout.QuadPart = (-120 * 1000 * 10000);
```

Comment:
Nit: Add a comment that this means 120 seconds and that we should make it configurable (impacts how long a user has to wait until the mapping is removed).